### PR TITLE
Added Note for the synchoronous Nature of the Web Storage API

### LINKED ISSUE #27686: Mention of the synchronous Nature of Web storage
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API

### What specific section or headline is this issue about?

https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#basic_concepts

### What information was incorrect, unhelpful, or incomplete?

The web Storage API methods are synchronous in Nature and a mention about it is important, as reading the Data from local Storage or even writing to it may take time and as it is synchronous it can affect the performance as the main thread operations would be consumed by it.

### What did you expect to see?

Few lines explaining the synchronous Nature of The Web Storage w

### Do you have any supporting links, references, or citations?

https://web.dev/storage-for-the-web/

clearly states in the localStorage section about it's synchronous Nature

### Do you have anything more you want to share?

_No response_

### MDN metadata

_No response_

### PR DESCRIPTION
**Pull request by Biki-das**
Fix #27686

I have added the info as a Note, if anyone feels they have a better way to add this , please let me know

cc @Josh-Cena

### CODE REVIEW

**Comment by Biki-das** [comment]
@Josh-Cena please review!

**Comment by Josh-Cena** [comment]
Sorry I'm not a WebAPI reviewer. You can just wait for your assigned reviewer.

**Comment by Biki-das** [comment]
@Elchi3 please review!

**Comment by Biki-das** [comment]
@teoli2003 could you give some feedback on this?

**Comment by Josh-Cena** [comment]
Please stop pinging people. Reviewers have priorities and they will get to yours when they have time.

**Comment by Biki-das** [comment]
> Please stop pinging people. Reviewers have priorities and they will get to yours when they have time.

😅sorry !

**Review by bsmth — CHANGES_REQUESTED** [review]
Thank you! A couple of comments:

* Let's move this to line 31 before the first note
* There are too many note blocks in a row here, let's use regular (2 or 3) paras.
* Consistently use backticks for `localStorage` etc.

**Comment by Biki-das** [comment]
@bsmth thanks for dropping the review, i will apply the requested changes as required

**Comment by Biki-das** [comment]
@bsmth  updated with the new changes

File: files/en-us/web/api/web_storage_api/index.md [hunk 1]
```diff
@@ -28,6 +28,12 @@ The two mechanisms within Web Storage are as follows:
...
 
+> **Note:** Both `sessionStorage` and `localStorage` in Web Storage are synchronous in nature. This means that when data is set, retrieved, or removed from these storage mechanisms, the operations are performed synchronously, blocking the execution of other JavaScript code until the operation is completed. This synchronous behavior can potentially affect the performance of the web application, especially if there is a large amount of data being stored or retrieved.
+
+> **Note:** Developers should be cautious when performing operations on `sessionStorage` or `localStorage` that involve a significant amount of data or computationally intensive tasks. It is important to optimize code and minimize synchronous operations to prevent blocking the user interface and causing delays in the application's responsiveness.
+
+> **Note:** Asynchronous alternatives, such as [IndexedDB](/en-US/docs/Web/API/IndexedDB_API), may be more suitable for scenarios where performance is a concern or when dealing with larger datasets. These alternatives allow for non-blocking operations, enabling smoother user experiences and better performance in web applications.
```
Comment (bsmth) [inline files/en-us/web/api/web_storage_api/index.md:35]:
```suggestion
Both `sessionStorage` and `localStorage` in Web Storage are synchronous in nature. This means that when data is set, retrieved, or removed from these storage mechanisms, the operations are performed synchronously, blocking the execution of other JavaScript code until the operation is completed. This synchronous behavior can potentially affect the performance of the web application, especially if there is a large amount of data being stored or retrieved.

Asynchronous alternatives, such as [IndexedDB](/en-US/docs/Web/API/IndexedDB_API), may be more suitable for scenarios where performance is a concern or when dealing with larger datasets. These alternatives allow for non-blocking operations, enabling smoother user experiences and better performance in web applications.
```

**Comment by bsmth** [comment]
> @bsmth thanks for dropping the review, i will apply the requested changes as required for the `There are too many note blocks in a row here, let's use regular (2 or 3) paras`. what i did is individually dividing and created note block is that what you want? could you brief on this .

Sorry, I meant we can use regular paragraphs instead of note (`> **note:**`) blocks. I've left a suggestion above for you, otherwise it's looking good to me, thanks!

**Comment by Biki-das** [comment]
@bsmth thanks for clarifying, i have pushed the changes!

**Review by bsmth — APPROVED** [review]
I'm happy with this in the current state. Thanks a lot for your patience and well done on the contribution!

**Pull request merged**